### PR TITLE
[ROCm][CI] skip float16 for TestTemplatedSDPA

### LINKED ISSUE #123531: DISABLED test_alibi_bias_float16 (__main__.TestTemplatedSDPA)
Platforms: ROCm

This test was disabled because it is failing on main branch ([recent examples](https://torch-ci.com/failure?failureCaptures=%5B%22inductor%2Ftest_templated_attention.py%3A%3ATestTemplatedSDPA%3A%3Atest_alibi_bias_float16%22%5D)).

cc @jeffdaily @sunway513 @jithunnair-amd @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### PR DESCRIPTION
**Pull request by jeffdaily**
Fixes #123531 and #123610 by replacing DISABLED issues with code skips.

cc @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by atalman — APPROVED** [review]

File: test/inductor/test_templated_attention.py [hunk 1]
```diff
@@ -31,6 +31,9 @@ def create_attention(score_mod):
     else [torch.float16, torch.float32]
 )
 
+# TODO float16 was causing ERRORs for tests on ROCm
```
Comment (atalman) [inline test/inductor/test_templated_attention.py:34]: Could you please include issue number here.

Comment (jeffdaily) [inline test/inductor/test_templated_attention.py:34]:
```suggestion
# TODO float16 was causing ERRORs for tests on ROCm
# See https://github.com/pytorch/pytorch/issues/123531
```

**Pull request closed without merging**